src/utilities/texture_atlas.jl

```diff
@@ -451,6 +451,103 @@ function primitive_uv_offset_width(atlas::TextureAtlas, marker::Observable, font
+        texture_atlas = Texture(eltype(images[1]), (max_xy...,))
+        for (area, img) in zip(uv_coordinates, images)
+            texture_atlas[area] = img # transfer to texture atlas
+        end
+        uvs = map(uv_coordinates) do uv
+            m = max_xy .- 1
+            mini = reverse((minimum(uv)) ./ m)
+            maxi = reverse((maximum(uv) .- 1) ./ m)
+            return Vec4f(mini..., maxi...)
+        end
+        images = texture_atlas
+    else
+        uvs = Vec4f(0,0,1,1)
+    end
+
+    return (uvs, images)
+end
+
+# For switching between ellipse method and faster circle method in shader
+is_all_equal_scale(o::Observable) = is_all_equal_scale(o[])
+is_all_equal_scale(::Real) = true
+is_all_equal_scale(::Vector{Real}) = true
+is_all_equal_scale(v::Vec2f) = v[1] == v[2] # could use ≈ too
+is_all_equal_scale(vs::Vector{Vec2f}) = all(is_all_equal_scale, vs)
+
+function compute_marker_attributes((atlas, uv_off, m, f, scale), changed, last)
+    # TODO, only calculate offset if needed
+    # [atlas_sym, :uv_offset_width, :marker, :font, :markersize]
+    # [:sdf_marker_shape, :sdf_uv, :image]
+    if m isa Matrix{<: Colorant} # single image marker
+        return (Cint(RECTANGLE), Vec4f(0,0,1,1), m)
+    elseif m isa Vector{<: Matrix{<: Colorant}} # multiple image markers
+        # TODO: Should we cache the RectanglePacker so we don't need to redo everything?
+        if changed.marker
+            uvs, images = pack_images(m)
+            return (Cint(RECTANGLE), uvs, images)
+        else
+            # if marker is up to date don't update
+            return (nothing, nothing, nothing)
+        end
+    else # Char, BezierPath, Vectors thereof or Shapes (Rect, Circle)
+        if changed.marker || changed.markersize
+            shape = Cint(Makie.marker_to_sdf_shape(m)) # expensive for arrays with abstract eltype?
+            if shape == 0 && !is_all_equal_scale(scale)
+                shape = Cint(5)
+            end
+        else
+            shape = last.sdf_marker_shape
+        end
+
+        if (shape == Cint(DISTANCEFIELD)) && (changed.marker || changed.font)
+            uv = Makie.primitive_uv_offset_width(atlas, m, f)
+        elseif isnothing(last)
+            uv = Vec4f(0,0,1,1)
+        else
+            uv = nothing # Is this even worth it?
+        end
+        return (shape, uv, nothing)
+    end
+end
+
+function all_marker_computations!(attr, atlas_res=1024, atlas_ppg=32)
+    atlas_sym = Symbol("atlas_$(atlas_res)_$(atlas_ppg)")
+    if !haskey(attr, atlas_sym)
+        register_computation!(attr, Symbol[], [atlas_sym]) do _, changed, last
+            (get_texture_atlas(atlas_res, atlas_ppg),)
+        end
+    end
+    inputs = [atlas_sym, :uv_offset_width, :marker, :font, :markersize]
+    outputs = [:sdf_marker_shape, :sdf_uv, :image]
+    register_computation!(
+        compute_marker_attributes, attr, inputs, outputs
+    )
+end
```

Comment:
When switching between GLMakie and WGLMakie this should error because we'll try to re-register the computation with "atlas_1024_32" instead of "atlas_2048_64" or vice versa. If it doesn't this may cause incorrect uvs to be used because they may be sampled from a different atlas.